### DISCUSSION TITLE
Customize Leaf's splash page

### DISCUSSION THREAD

**Discussion by artgoldberg** [uwrit/leaf#578] [Ideas]
My Dean raises an issue that I think would apply to multiple Leaf installations. She would like to display Mount Sinai’s contributing organizations and grants on Leaf’s splash page.

![image](https://user-images.githubusercontent.com/33882/200926550-8b0ad22a-296f-42d9-ada3-c25edc541647.png)

We have two organizations and one grant she’d like to add.
 
Here’s my brainstorm: Leaf's organizations and paper should continue to be displayed, in my opinion. 

Our feature request could be configurable. One or more logos and the text for a grant (or for anything, like “Customized for Mount Sinai by Prof. Arthur Goldberg.”) could optionally be added to the app settings file, and if present would be displayed. Perhaps this should appear on a “credits” page with just a Next button, which gets displayed before this page of user properties.

**Comment by ndobb** [comment]
Thanks @artgoldberg. That makes sense. I'm thinking visually we could put some space between the top "Leaf ... ITHS ... CD2H" and the "I want to find information for: QI ... Research" for a custom logo and text. A credits page is an option as well, but I'd like to save users a click if we can. We might need to experiment a bit.

I agree that configuring the custom text via the app settings file makes sense. For logos it's a bit trickier, as - assuming a site has added their logo picture file(s) somewhere in their production Apache web server - we would need some way to allow Leaf to know the paths of the logos it should show. To my mind we could do this by

1. Including the relative paths to logos also in the app settings file, *or*
2. Adding a new folder for deployment (eg, `/build/styles/custom_logos`), for which the app will simply show all picture files included in.

(1) seems more straightforward to me. Does this all sound reasonable, @artgoldberg?

Best,
-nic

**Comment by artgoldberg** [comment]
Thanks @ndobb that does sound reasonable, with one caveat.

If you also want to minimize user clicks, I have another argument for separating a new "credits" page from the page in the screenshot above (I'll call it the _user status and data_ page). I had planned to avoid adding this complication to this update, but since you raise the issue of user clicks I feel compelled to mention it.

In our installation, Leaf only access as de-identified database. And if we wanted to configure it to access a PHI database, we would deploy another Leaf instance. I think Johns Hopkins has the same setup.
We do this because we implement de-identification at the database level, by creating a de-identified view of the PHI data. In fact, we have many views, because some users only want (or are permitted) to see subsets of the data. Therefore, we might deploy many Leaf instances, each of which has a particular predetermined user status and data set.

Therefore, none of the data on the _user status and data_ page is needed by our deployment, and our user sign-in experience would be greatly improved by removing all of the clicks on it.

To summarize, I propose that Leaf's configuration in `appsettings` be changed to have:

1. A _credits_ page, with your credits and the ability for installations to add their own. It could be displayed for a brief time (with a configurable timeout) to avoid a click.
2. An optional  _user status and data_ page.

What do you and others think?

Thanks!
Arthur

**Comment by ndobb** [comment]
Hi @artgoldberg , thanks very much for your patience. I'm working through the design for this and wanted to get your thoughts.

Regarding the credits page, I'm wondering if it would be sufficient to simply append the logos and institutional message below the existing Leaf, ITHS, and CD2H logos at the top login, like this:

![image](https://user-images.githubusercontent.com/10424962/223214025-51926180-2ac9-4499-a98d-2daec7998242.png)

Configuration:

```json
"Attestation": {
    "Enabled": true,
    "SkipModeSelection": false,
    "Credits": {
      "Enabled": true,
      "Logos": [
        "./images/splash/mtsinai.png"
      ],
      "Text": "Customized for Mount Sinai by Prof. Arthur Goldberg"
    }
  },
```

Adding additional logos (using UW here as an example) automatically resizes the row to evenly distribute space:

![image](https://user-images.githubusercontent.com/10424962/223215600-1452cc54-0ef5-4d67-bf63-7d3484930177.png)

```json
"Attestation": {
      "Logos": [
        "./images/splash/mtsinai.png",
        "./images/splash/uw.jpg"
      ],
  },
```

Last, setting `SkipModeSelection` to `true` automatically skips the user status and data page (it defaults to false if not specified) and goes straight to the attestation.

```json
"Attestation": {
    "SkipModeSelection": true,
  },
```

![image](https://user-images.githubusercontent.com/10424962/223216317-13598b5d-438b-4e09-a5eb-66e59d071c68.png)

Sorry for all the pics - hopefully the above makes some sense.

My hope is that the flexibility offered here should be valuable to different sites for different use cases, and sites can mix & match custom attestations, custom logos and credits, skipping the Research vs. QI selection, and so on.

What do you think?

Best,
-nic

**Reply by artgoldberg** [reply]
Thanks @ndobb . Personally, I think it looks great. I'm checking with my dean who requested this and will let you know what I hear.

**Reply by artgoldberg** [reply]
Hi @ndobb 

Sorry to take so long to reply. This looks like a great way to customize Leaf's splash page.

Arthur
